app/include/drivers/behavior.h

```diff
@@ -23,6 +23,64 @@
  * (Internal use only.)
  */
 
+#if IS_ENABLED(CONFIG_ZMK_BEHAVIOR_METADATA)
+
+enum behavior_parameter_standard_domain {
+    BEHAVIOR_PARAMETER_STANDARD_DOMAIN_NULL = 0,
+    BEHAVIOR_PARAMETER_STANDARD_DOMAIN_HID_USAGE = 1,
+    BEHAVIOR_PARAMETER_STANDARD_DOMAIN_LAYER_INDEX = 2,
+    BEHAVIOR_PARAMETER_STANDARD_DOMAIN_HSV = 3,
+};
+
+struct behavior_parameter_value_metadata {
+    char *friendly_name;
+    uint8_t position;
+
+    union {
+        uint32_t value;
+        struct {
+            uint16_t min;
+            uint16_t max;
+        } range;
```

Comment:
Hmm.... some good questions. I suppose for this, I was having a hard time imagining a range needing to be larger than this, or negative, but since we're trying to have this not have to change any time soon.... worth being a bit more flexible here.

Do we think `{ int32_t min; uint32_t max; } range` would work? Do we *really* think we need a range that's only negative numbers? or with a minimum above 2,147,483,647 ever?

Reply:
Mixing signed and unsigned math in C is generally a great way to have a bad time. I'd be in favor of just using signed values for both min and max. I can't think of any real-world range of values that wouldn't fit into a signed 32-bit integer.

The first thing I could think of that might need large numbers is millisecond durations. With 16-bit, that could be a problem since you'd be limited to ~32 s signed or ~65 s unsigned. With signed 32-bit, you get ~25 days, which I can't imagine being insufficient for anything (if that's the kind of time scale your behavior is working on, you should probably be using seconds instead of milliseconds).

Reply:
Yeah, totally fair. I'll update for signed 32-bit values. It'll increase the size a small bit, but not enough I think to warrant limiting things.